src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,55 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
+
+  static removeAnimal(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
+
+  static getAliveAnimals() {
+    return Animal.alive.map((animal) => animal.toObject());
+  }
+
+  toObject() {
+    return { name: this.name, health: this.health };
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  toObject() {
+    return { ...super.toObject(), hidden: this.hidden };
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        target.health = 0;
+        Animal.removeAnimal(target);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] - Instead of checking the health of the target animal and then removing it inside the `bite` method, it's better to use the `filter` method to remove any dead animals from the `Animal.alive` array. This would avoid the need for an explicit check and removal within the `bite` method and ensures that dead animals are cleaned up in a single location.